xchange-poloniex2/src/main/java/info/bitrich/xchangestream/poloniex2/PoloniexStreamingMarketDataService.java

```diff
@@ -75,10 +81,27 @@ public PoloniexStreamingMarketDataService(PoloniexStreamingService service, Map<
     @Override
     public Observable<Trade> getTrades(CurrencyPair currencyPair, Object... args) {
         Observable<PoloniexWebSocketTradeEvent> subscribedTrades = service.subscribeCurrencyPairChannel(currencyPair)
+                .flatMapIterable(poloniexWebSocketEvents -> poloniexWebSocketEvents)
                 .filter(s -> s.getEventType().equals("t"))
```

Comment:
Might be slightly clearer if this were:
```
.filter(PoloniexWebSocketTradeEvent.class::isInstance)
```
Since we've gone to the effort of assigning these types beforehand.